internal/raft/append_entries.go

```diff
@@ -6,6 +6,50 @@ import "github.com/tomarrell/lbadd/internal/raft/message"
 // to the follower node. This function generates the response to be sent to the leader node.
 // This is the response to the contact by the leader to assert it's leadership.
 func AppendEntriesResponse(node *Node, req *message.AppendEntriesRequest) *message.AppendEntriesResponse {
+	success := true
+	leaderTerm := req.GetTerm()
+	nodePersistentState := node.PersistentState
+	nodeTerm := nodePersistentState.CurrentTerm
+	if nodeTerm > leaderTerm {
+		success = false
+	} else if req.GetPrevLogIndex() > int32(node.VolatileState.CommitIndex) {
+		success = false
+	} else if nodePersistentState.Log[req.PrevLogIndex].Term != req.GetPrevLogTerm() {
+		success = false
+	}
+
+	if !success {
+		return &message.AppendEntriesResponse{
+			Term:    nodeTerm,
+			Success: success,
+		}
+	}
+
+	entries := req.GetEntries()
+	if len(entries) > 0 { // if heartbeat, skip adding entries
+		nodePersistentState.mu.Lock()
+		if req.GetPrevLogIndex() < int32(node.VolatileState.CommitIndex) {
+			node.PersistentState.Log = node.PersistentState.Log[:req.GetPrevLogIndex()]
+		}
+		for _, entry := range entries {
+			node.PersistentState.Log = append(node.PersistentState.Log, *entry)
```

Comment:
@TimSatke , the pointer is being passed here right? Im not sure whats wrong here, can you take a look?

Reply:
Which pointer?

Reply:
*entry pointer passed to append function

Reply:
Lint fails saying lock value is copied.

Reply:
you are dereferencing the pointer, meaning you explicitely are using a value